include/rtosc/bundle-foreach.h

```diff
@@ -31,53 +31,75 @@
 #ifndef BUNDLE_FOREACH
 #define BUNDLE_FOREACH
 
+#include <cassert>
 #include <cctype>
 #include <cstdlib>
 #include <cstdio>
 #include "ports.h"
 
 namespace rtosc {
 
+/*
+                          Ports& base
+                               |p.name
+                               vv
+               "/path/from/root/new#2/path#2/to#2/"
+
+               "/path/from/root/new1/path1/"
+                ^               ^
+   name_buffer[buffer_size]     old_end
+
+*/
 /**
  * Execute a callback for all bundle elements of a bundle port
+ * @param p the bundle port
  * @param name Should be p.name
+ * @param old_end points at end of name_buffer
+ * @param name_buffer functor-data: buffer containing the path until and including `base`
+ * @param base functor-data: base Ports struct containing `p`
+ * @param data functor-data: arbitrary data for the functor
+ * @param runtime functor-data: the OSC runtime object
+ * @param ftor the functor to be called for each port of the bundle
+ * @param expand_bundles if false, don't iterate, just print name without numbers
+ * @param cut_afterwards if true, before returning, the string is cut at old_end
+ * @param ranges don't iterate, just print `[0,max]`
+ *
  * TODO: pass base as pointer, so it can be null?
  */
 template<class F>
-void bundle_foreach(const struct Port& p, const char* name, char* old_end,
+void bundle_foreach(const struct Port& p, const char* name, char* const old_end,
+                    const char* const name_buffer, size_t buffer_size,
                     /* data which is only being used by the ftors */
-                    const char* name_buffer, const struct Ports& base,
-                    void* data, void* runtime, const F& ftor,
+                    const struct Ports& base,
+                    void* const data, void* const runtime, const F& ftor,
                     /* options */
-                    bool expand_bundles = true,
-                    bool cut_afterwards = true,
-                    bool ranges = false)
+                    const bool expand_bundles = true,
+                    const bool cut_afterwards = true,
+                    const bool ranges = false)
 {
+    size_t space_left = buffer_size - (size_t)(old_end - name_buffer);
```

Comment:
minor, but I'd use ssize_t here to keep things signed in this computation in case there is a buggy old_end/name_buffer input (aka to handle buggy cases where space_left would be negative).

Reply:
Fixed by using `ssize_t`.